src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -23,7 +24,7 @@ class GolangSubsystem(Subsystem):
     options_scope = "golang"
     help = "Options for Golang support."
 
-    class EnvironmentAware(Subsystem.EnvironmentAware):
+    class EnvironmentAware(ExecutableSearchPathsOptionMixin, Subsystem.EnvironmentAware):
         env_vars_used_by_options = ("PATH",)
 
         _go_search_paths = StrListOption(
```

Comment:
@tdyas with the addition of the `ExecutableSearchPathsOptionMixin`, this search path and others like the one for cgo_tool could be removed. However, this would be a breaking change for existing configs. What are your thoughts?

Reply:
Seems like something to consider in a separate PR. This PR should just focus on the `--golang-extra-tools` option.

Reply:
That said, I'd have to review on what exactly`ExecutableSearchPathsOptionMixin` does and what its other uses are.  Better done in a separate PR (or start off discussion in an issue).

Reply:
@tdyas I've created a separate issue https://github.com/pantsbuild/pants/issues/21346